src/multiple_edgetpu_demo.cc

```diff
@@ -119,56 +121,82 @@ class AverageCounter {
   std::deque<double> data_;
 };
 
-class GstWrapperAllocator : public coral::Allocator {
+class DmaBuffer : public coral::Buffer {
  public:
-  GstWrapperAllocator() = default;
+  DmaBuffer(GstSample *sample, size_t requested_bytes)
+      : sample_(CHECK_NOTNULL(sample)), requested_bytes_(requested_bytes) {}
+
+  // For cases where we can't use DMABuf (like most x64 systems), return the
+  // sample. This allows the pipeline runner to see there is no file descriptor
+  // and instead just pass the sample as the input tensor.
+  void *ptr() override { return sample_; }
```

Comment:
I think we still need to get the GstBuffer from GstSample, and then get the data from GstBuffer. Similar to what we did before. First field of GstBuffer is not the image data
https://github.com/Kurento/gstreamer/blob/master/gst/gstsample.c#L37


    GstMapInfo info;	
    GstBuffer *buffer = CHECK_NOTNULL(gst_sample_get_buffer(sample));	
    CHECK(gst_buffer_map(buffer, &info, GST_MAP_READ));	
    void *data = reinterpret_cast<void *>(info.data);